storage/connect/connect.cc

```diff
@@ -546,7 +546,8 @@ int CntCloseTable(PGLOBAL g, PTDB tdbp, bool nox, bool abort)
 	if (!tdbp)
 		return rc;                           // Nothing to do
 	else if (tdbp->GetUse() != USE_OPEN) {
-		if (tdbp->GetAmType() == TYPE_AM_XML)
+		if (tdbp->GetAmType() == TYPE_AM_XML ||
+			tdbp->GetAmType() == TYPE_AM_JSN)
```

Comment:
Seems that this happens on server shutdown. 
We are missing minimal test case to reproduce this. 
I suggest to add the minimal test case in `json.test`, referencing MDEV-29426, so we could have a proper trace of MDEV.
Could this be solved with proper destructor for class `TDBJSN`, since is inherited from `TDB` class ?

Reply:
> Seems that this happens on server shutdown. We are missing minimal test case to reproduce this. I suggest to add the minimal test case in `json.test`, referencing MDEV-29426, so we could have a proper trace of MDEV. Could this be solved with proper destructor for class `TDBJSN`, since is inherited from `TDB` class ?

As per the PR description, the current JSON and BSON test cases reproduce this with ASAN enabled (or Valgrind).

`TDBJSN` isn't the problem, `JSONDEF` is, and `TDBJSN`'s close cleans it up properly.

Reply:
After a call with an3l it was decided that I would add an additional specific test for this.

Reply:
Test cases added.